libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -888,6 +893,23 @@ bool AP_Arming::servo_checks(bool report) const
             }
         }
     }
+#if HAL_WITH_IO_MCU && (APM_BUILD_COPTER_OR_HELI || APM_BUILD_TYPE(APM_BUILD_ArduPlane) || APM_BUILD_TYPE(APM_BUILD_Rover))
+#if APM_BUILD_TYPE(APM_BUILD_Rover)
+    AP_MotorsUGV *motors = AP::motors_ugv();
+#else
+    AP_Motors *motors = AP::motors();
+#endif
+    if (motors != nullptr && motors->is_digital_pwm_type())) {
+        uint32_t mask = motors->get_motor_mask();
+        for (uint8_t i = 0; i < 8; i++) {
+            if (mask & 1U<<i) {
+                SRV_Channel::Aux_servo_function_t function = SRV_Channels::get_motor_function(i);
+                check_failed(report, "SERVO%u_FUNCTION set to Motor%u on non-digital channel", unsigned(function), i + 1);
```

Comment:
I think you should mention the PWM type param. One way to fix is to move the motor output, the other way is to stop asking for a digital PWM type.

Would be nice to also do the check for the overide type on none motor ouptuts.